src/components/SlugField.tsx

```diff
@@ -0,0 +1,62 @@
+import { Stack, Text } from '@sanity/ui';
+import { Reference, SlugInput, SlugInputProps, SlugValue, useEditState, useFormValue } from 'sanity';
+import { PageTreeConfig } from '../types';
+import { usePageTreeItem } from '../hooks/usePageTreeItem';
+import { getSanityDocumentId } from '../utils/sanity';
+
+export type SlugFieldProps = {
+  config: PageTreeConfig;
+} & SlugInputProps;
+
+export const SlugField = (props: SlugFieldProps) => {
+  const id = useFormValue(['_id']);
+  const type = useFormValue(['_type']);
+  // eslint-disable-next-line no-warning-comments
+  // TODO ideally this would be more type safe.
+  const parentRef = useFormValue(['parent']) as Reference | undefined;
+
+  const { config, value, ...otherProps } = props;
+
+  return (
+    <Stack space={3}>
+      <SlugInput {...otherProps} value={value} />
```

Comment:
I think it is better to use the `props.renderDefault` function here instead of manually importing and using the `SlugInput` component from sanity.